pkg/util/iptables/iptables.go

```diff
@@ -49,6 +51,18 @@ type Interface interface {
 	DeleteRule(table Table, chain Chain, args ...string) error
 	// IsIpv6 returns true if this is managing ipv6 tables
 	IsIpv6() bool
+	// Save calls `iptables-save` for table.
+	Save(table Table) ([]byte, error)
```

Comment:
Ah yes, one callsite that mocks it in the userspace proxier tests, will
patch that in shortly. Forgot that, quick-release built fine, tests break
from this.

On Fri, Aug 7, 2015 at 6:09 PM, Tim Hockin <notifications@github.com> wrote:

> In pkg/util/iptables/iptables.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12416#discussion_r36565795>
> :
>
> > @@ -49,6 +51,18 @@ type Interface interface {
> >  	DeleteRule(table Table, chain Chain, args ...string) error
> >  	// IsIpv6 returns true if this is managing ipv6 tables
> >  	IsIpv6() bool
> > +	// Save calls `iptables-save` for table.
> > +	Save(table Table) ([]byte, error)
>
> Does the addition of these methods break any other call-sites that assume
> they are implementing iptables.Interface ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12416/files#r36565795>
> .
>